internal/venice-common/src/main/java/com/linkedin/venice/pubsub/manager/TopicMetadataFetcher.java

```diff
@@ -183,9 +183,14 @@ void validateTopicPartition(PubSubTopicPartition pubSubTopicPartition) {
     if (pubSubTopicPartition.getPartitionNumber() < 0) {
       throw new IllegalArgumentException("Invalid partition number: " + pubSubTopicPartition.getPartitionNumber());
     }
-    if (containsTopicCached(pubSubTopicPartition.getPubSubTopic())) {
-      return;
+    try {
+      if (containsTopicCached(pubSubTopicPartition.getPubSubTopic())) {
+        return;
+      }
+    } catch (Exception e) {
+      // in case there is any exception, we will retry the operation
     }
```

Comment:
log the exception here. It doesn't look like a highly perf sensitive code path as we already have cache and unless there is something seriously wrong with the fetcher flow.

IMO, The retries without any information on why we do retry and how long is going to be hard to debug.

Reply:
>log the exception here. It doesn't look like a highly perf sensitive code path as we already have cache and unless there is something seriously wrong with the fetcher flow.

I don't think this is good idea since this method is called frequently. 


>as we already have cache and unless there is something seriously wrong with the fetcher flow.
Could you please expand this?

Reply:
>Could you expand this?

I still don't have full context. But glancing `TopicMetadataFetcher` and `TopicManager`, it seems some callers of this method `validateTopic...` themselves have some sort of retries. That made me come to conclusion that on a happy path, you wouldn't log anything and on a failure scenario, you would log the exception and the cadence of log spam would be tied the retry mechanisms employed by the callers (which is typically, exponential backoff) and thus preventing log spam.

Reply:
Your observation is valid; however, the number of failures can often be significant. This is because each node hosts multiple partitions, and this code path is called several times a minute for these partitions, either for metrics collection or other internal bookkeeping. When things go wrong it doesn't take a long time to fill up logs with redundant info.